xtask/src/main.rs

```diff
@@ -0,0 +1,336 @@
+                                    );
+                                };
+                            let language = language.strip_suffix('\n').unwrap_or(language);
+
+                            let file = open_file(path)?;
+                            EasyCommand::new(xcrun, |cmd| {
+                                cmd.stdin(Stdio::from(file))
+                                    .args(["-sdk", "macosx", "metal", "-mmacosx-version-min=10.11"])
+                                    .arg(format!("-std=macos-{language}"))
+                                    .args(["-x", "metal", "-", "-o", "/dev/null"])
+                            })
+                            .success()
+                        },
+                    )
+                }
+                ValidateSubcommand::Glsl => {
+                    let glslang_validator = "glslangValidator";
+                    which(glslang_validator)?;
+                    for (glob, type_arg) in [
+                        ("glsl/*.Vertex.glsl", "vert"),
+                        ("glsl/*.Fragment.glsl", "frag"),
+                        ("glsl/*.Compute.glsl", "comp"),
+                    ] {
+                        visit_files(&snapshots_base_out, glob, &mut found_err, |file, _| {
+                            log::info!("Validating {}", file.display());
+                            let file = open_file(file)?;
+                            EasyCommand::new(glslang_validator, |cmd| {
+                                cmd.stdin(Stdio::from(file))
+                                    .args(["--stdin", "-S"])
+                                    .arg(type_arg)
+                            })
+                            .success()
+                        });
+                    }
+                }
+                ValidateSubcommand::Dot => {
+                    let dot = "dot";
+                    which(dot)?;
+                    visit_files(
+                        snapshots_base_out,
+                        "dot/*.dot",
+                        &mut found_err,
+                        |file, _| {
+                            // TODO: de-dupe this `log::info!(…)`
+                            log::info!("Validating {}", file.display());
+                            let file = open_file(file)?;
+                            EasyCommand::new(dot, |cmd| {
+                                cmd.stdin(Stdio::from(file)).stdout(Stdio::null())
+                            })
+                            .success()
+                        },
+                    )
+                }
+                ValidateSubcommand::Wgsl => visit_files(
+                    snapshots_base_out,
+                    "wgsl/*.wgsl",
+                    &mut found_err,
+                    |file, _| {
+                        // TODO: strip prefix to match old Makefile output?
+                        log::info!("Validating {}", file.display());
```

Comment:
N.B. that this output is technically different from the original `Makefile`; it inclues the path prefix of the glob search path, instead of "just" printing the base name. I don't think anybody will care, tho!